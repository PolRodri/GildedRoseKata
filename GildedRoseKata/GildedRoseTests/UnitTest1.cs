using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using csharp;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace GildedRoseTests
{
    [UseReporter(typeof(DiffReporter))]
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethod1()
        {
            String[] name = { "foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert" };
            int[] sellin = { 0 , -1, 5};
            int[] quality = { 0, 1, 2 };

            //string itemString = doUpdateQuality(name, sellin, quality);

            //Assert.AreEqual("foo", items[0].Name);

            //Approvals.Verify(itemString);
            CombinationApprovals.VerifyAllCombinations(doUpdateQuality,  name, sellin, quality);
        }

        private static string doUpdateQuality(string name, int sellin, int quality)
        {
            IList<Item> items = new List<Item> { new Item { Name = name, SellIn = sellin, Quality = quality } };
            GildedRose gildedRose = new GildedRose(items);

            gildedRose.UpdateQuality();

            String itemString = items[0].ToString();
            return itemString;
        }
    }
}
