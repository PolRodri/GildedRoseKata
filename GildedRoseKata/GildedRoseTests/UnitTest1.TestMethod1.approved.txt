[foo,0,0] => foo, -1, 0
[foo,0,1] => foo, -1, 0
[foo,0,2] => foo, -1, 0
[foo,-1,0] => foo, -2, 0
[foo,-1,1] => foo, -2, 0
[foo,-1,2] => foo, -2, 0
[foo,5,0] => foo, 4, 0
[foo,5,1] => foo, 4, 0
[foo,5,2] => foo, 4, 1
[Aged Brie,0,0] => Aged Brie, -1, 2
[Aged Brie,0,1] => Aged Brie, -1, 3
[Aged Brie,0,2] => Aged Brie, -1, 4
[Aged Brie,-1,0] => Aged Brie, -2, 2
[Aged Brie,-1,1] => Aged Brie, -2, 3
[Aged Brie,-1,2] => Aged Brie, -2, 4
[Aged Brie,5,0] => Aged Brie, 4, 1
[Aged Brie,5,1] => Aged Brie, 4, 2
[Aged Brie,5,2] => Aged Brie, 4, 3
[Backstage passes to a TAFKAL80ETC concert,0,0] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,1] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,2] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,1] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,2] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,5,0] => Backstage passes to a TAFKAL80ETC concert, 4, 3
[Backstage passes to a TAFKAL80ETC concert,5,1] => Backstage passes to a TAFKAL80ETC concert, 4, 4
[Backstage passes to a TAFKAL80ETC concert,5,2] => Backstage passes to a TAFKAL80ETC concert, 4, 5
